src/main/java/Main.java

```diff
@@ -414,6 +420,30 @@ private static File extractFromJar(String resource, String fileName, String suff
         return tmp;
     }
 
+    private static void deleteWinstoneTempContents(File folder, final String...patterns) throws IOException {
+        File tmpdir = folder;
+
+        if(tmpdir == null){
+            tmpdir = new File(System.getProperty("java.io.tmpdir"));
+        }
+
+        final File[] files = tmpdir.listFiles(new FilenameFilter() {
```

Comment:
This method may return `null` if there is an I/O Error: https://docs.oracle.com/javase/7/docs/api/java/io/File.html#listFiles(java.io.FileFilter)

Reply:
I protected the NPE before the `for` sentence